pandas/_libs/groupby.pyx

```diff
@@ -778,47 +778,48 @@ def group_quantile(ndarray[float64_t] out,
             if not mask[i]:
                 non_na_counts[lab] += 1
 
-    # Get an index of values sorted by labels and then values
-    order = (values, labels)
-    sort_arr = np.lexsort(order).astype(np.int64, copy=False)
-
-    with nogil:
-        for i in range(ngroups):
-            # Figure out how many group elements there are
-            grp_sz = counts[i]
-            non_na_sz = non_na_counts[i]
+    if labels.any():
+        # Get an index of values sorted by labels and then values
+        labels[labels==-1] = np.max(labels) + 1
```

Comment:
I think rather than nesting this block in an if statement we just want to avoid taking the maximum of an empty array (which we can do without touching most of this function), the subsequent for loop should do nothing in that case anyways

Reply:
Totally agree, I was just fixing that. Just doing the operation if there any labels.